.github/workflows/merge-pr.yml

```
@@ -6,16 +6,42 @@ on:
   push:
     branches:
       - main
-    paths:
-      - "js/api-augment/**"
-      - "**/*.rs"
-      - "**/Cargo.toml"
-      - "**/Cargo.lock"
-env:
-  BUILD_PROFILE: release
-
 jobs:
+  changes:
+    name: Determine Changed Files
+    runs-on: ubuntu-20.04
+    container: public.ecr.aws/lts/ubuntu:20.04
+    outputs:
+      rust: ${{steps.filter.outputs.rust}}
+      js-api-augment: ${{steps.filter.outputs.js-api-augment}}
+      ci-base-image: ${{steps.filter.outputs.ci-base-image}}
+    steps:
+      - name: Install Required Packages
+        run: |
+          apt-get update
+          apt-get install -y git
+          git config --global --add safe.directory /__w/frequency/frequency
+      - name: Check Out Repo
+        uses: actions/checkout@v3
+      - name: Check for Changed Files
+        uses: dorny/paths-filter@4512585405083f25c027a35db413c2b3b9006d50
+        id: filter
+        with:
+          filters: |
+            rust:
+              - '**/*.rs'
+              - '**/Cargo.toml'
+              - '**/Cargo.lock'
+            js-api-augment:
```

I like having a build of api-augment for each sha, but it isn't important. However, it needs to also run if Rust changes because it generates code off of the rust.

```suggestion
            js-api-augment:
              - '**/*.rs'
```